src/libstore/gc-store.hh

```diff
@@ -9,52 +9,35 @@ namespace nix {
 
 typedef std::unordered_map<StorePath, std::unordered_set<std::string>> Roots;
 
+/* Return either live (reachable) or dead (unreachable) paths */
+enum class GCReturn { Live, Dead };
 
-struct GCOptions
-{
-    /**
-     * Garbage collector operation:
-     *
-     * - `gcReturnLive`: return the set of paths reachable from
-     *   (i.e. in the closure of) the roots.
-     *
-     * - `gcReturnDead`: return the set of paths not reachable from
-     *   the roots.
-     *
-     * - `gcDeleteDead`: actually delete the latter set.
-     *
-     * - `gcDeleteSpecific`: delete the paths listed in
-     *    `pathsToDelete`, insofar as they are not reachable.
-     */
-    typedef enum {
-        gcReturnLive,
-        gcReturnDead,
-        gcDeleteDead,
-        gcDeleteSpecific,
-    } GCAction;
+/* Set of paths to delete, and whether to skip paths which are alive */
+struct GCPathsToDelete {
+   StorePathSet paths;
+   bool skipAlive;
+};
 
-    GCAction action{gcDeleteDead};
+/* Delete either a given set of paths, or all dead paths */
```

Comment:
Please continue to use `/**` style comments so it renders in the internal API docs.